app/models/workday/commercial_agreements.rb

```diff
@@ -0,0 +1,38 @@
+class Workday::CommercialAgreements
+  def revenue_category_ids
+    result = {}
+    report_entries.each do |report_entry|
+      framework_number = report_entry.at_xpath('wd:Framework_Number').text
+      revenue_category_wid_xml = report_entry.at_xpath('wd:Revenue_Category_ID')
+      result[framework_number] = revenue_category_wid_xml.text if revenue_category_wid_xml
+    end
+    result
+  end
+
+  def tax_code_ids
+    result = {}
+    report_entries.each do |report_entry|
+      framework_number = report_entry.at_xpath('wd:Framework_Number').text
+      tax_code_xml = report_entry.at_xpath('wd:Tax_Code_Reference/wd:ID[@wd:type="commercialAgreementTaxCode"]')
+      result[framework_number] = tax_code_xml.text if tax_code_xml
+    end
+    result
+  end
+
+  private
+
+  def report_entries
+    @report_entries ||= Nokogiri(commercial_agreement_xml).xpath('//wd:Report_Entry')
+  end
+
+  def commercial_agreement_xml
+    @commercial_agreement_xml ||= HTTP.basic_auth(
+      user: Workday.username,
+      pass: Workday.api_password
+    ).get(
+      'https://wd3-impl-services1.workday.com/ccx/service/customreport2/' +
```

Comment:
Is this host fixed or is it one possible load-balanced endpoint of several? Could we factor bits of this URL into somewhere constant-y or configure-y, say `Workday::BASE_URL` or a `workday.base_url = <config_value>` or somesuch? (don't know enough about rest of Workday to know)

Reply:
I'm not quite sure of the internals, but this API endpoint is different from the main SOAP one, which is defined in the WSDL. It's only used here and it different to how we normally interact with Workday.

Reply:
Ah, ok. Do we need to say `Workday::UNDOCUMENTED_THING_THEY_TOLD_US_TO_USE` or somesuch?